### ISSUE TITLE
Manager interfering with ReactMeteorData

### ISSUE THREAD

**Issue by tomitrescak** [kadirahq/subs-manager#55]
Hi, first thanks for this optimisation item without which I cannot my applications to be. I have found a possible bug in the implementation, when subscription manager is interfering with "ReactMeteorData" mixin. I am attaching a minimal repro application.

The problem is, that if you use a component with subscriptions inside another component with subscriptions, it is invalidating the parent subscriptions, making them run twice. Example:

I have following components and only parent has subscriptions:

<Parent>
  <Child>
<Parent>

If I put following line inside paren's "render" method `console.log("With: " + this.data.posts)` where data is fed with following method:

```javascript
getMeteorData() {
    this.data = { }
    var handle = subs.subscribe("posts");
    if (handle.ready()) {
      this.data = {
        posts: Posts.find().fetch()
      }
    }
    return this.data;
  },
``` 

The output is following as expected:

```
With: undefined
With: undefined
With: [object Object],[object Object],[object Object]
```

Now I add another subscription inside child component as following:

```javascript
  getMeteorData() {
    this.data = { }
    var handle = subs.subscribe("user", this.props.post.uid); // Replace with Meteor.subscribe and all is good
    if (handle.ready()) {
      this.data = {
        user: Usrs.findOne({uid: this.props.post.uid})
      }
    }

    return this.data;
  }
```

Output becomes:

```
With: undefined
With: undefined
With: [object Object],[object Object],[object Object]
With: undefined
With: [object Object],[object Object],[object Object]
```

Somewhere there, the subscription invalidated meteor data and it ran again creating havoc in the application. Please note, that when I replace `subs.subscribe` with `Meteor.subscribe` all is running as expected and data is not invalidated. 

Example minimal app is [here 6Kb](https://dl.dropboxusercontent.com/u/3418607/react-example.zip).

**Comment by arunoda** [comment]
Yes. I can understand the issue. I think it's hard to mitigate this unless, you create two instances of subsManagers to each component.

I've added some notes here: https://goo.gl/9PeI1Q

**Comment by tomitrescak** [comment]
Thanks! Just a question regarding performance .. .will that affect subsmanager performance if I will be creating a new instance of subs manager per each subscription? Will that interfere with its functionality ?
Thanks.

**Comment by arunoda** [comment]
Nope. It create caches separately.
